### PR TITLE
test: Fix osd-scrub-scaps.sh to handle DB format change

### PR DESCRIPTION
**Pull request by dzafman**
Caused by: f9c95fa7fc7b0ee992c0249ff090fa7f751e9719

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by neha-ojha — APPROVED** [review]

File: qa/standalone/scrub/osd-scrub-snaps.sh [hunk 1]
```diff
@@ -101,11 +101,13 @@ function create_scenario() {
     OBJ5SAVE="$JSON"
     # Starts with a snapmap
     ceph-kvstore-tool bluestore-kv $dir/${osd} list 2> /dev/null > $dir/drk.log
-    grep "^m.*SNA_.*[.]1[.]obj5[.][.]$" $dir/drk.log || return 1
+    grep SNA_ $dir/drk.log
```
Comment (ronen-fr) [inline qa/standalone/scrub/osd-scrub-snaps.sh:104]: is this just for logging?

File: qa/standalone/scrub/osd-scrub-snaps.sh [hunk 2]
```diff
@@ -101,11 +101,13 @@ function create_scenario() {
...
+    grep "^p.*SNA_.*[.]1[.]obj5[.][.]$" $dir/drk.log || return 1
     ceph-objectstore-tool --data-path $dir/${osd} --rmtype nosnapmap "$JSON" remove || return 1
     # Check that snapmap is stil there
     ceph-kvstore-tool bluestore-kv $dir/${osd} list 2> /dev/null > $dir/drk.log
-    grep "^m.*SNA_.*[.]1[.]obj5[.][.]$" $dir/drk.log || return 1
+    grep SNA_ $dir/drk.log
+    grep "^p.*SNA_.*[.]1[.]obj5[.][.]$" $dir/drk.log || return 1
```
Comment (ronen-fr) [inline qa/standalone/scrub/osd-scrub-snaps.sh:110]: I would suggest accepting both 'm' & 'p', as to not create a versioning issue

File: qa/standalone/scrub/osd-scrub-snaps.sh [hunk 3]
```diff
@@ -121,12 +123,14 @@ function create_scenario() {
 
     # Starts with a snapmap
     ceph-kvstore-tool bluestore-kv $dir/${osd} list 2> /dev/null > $dir/drk.log
-    grep "^m.*SNA_.*[.]7[.]obj16[.][.]$" $dir/drk.log || return 1
+    grep SNA_ $dir/drk.log
```
Comment (ronen-fr) [inline qa/standalone/scrub/osd-scrub-snaps.sh:126]: do we need this log?

File: qa/standalone/scrub/osd-scrub-snaps.sh [hunk 4]
```diff
@@ -121,12 +123,14 @@ function create_scenario() {
...
+    grep "^p.*SNA_.*[.]7[.]obj16[.][.]$" $dir/drk.log || return 1
     JSON="$(ceph-objectstore-tool --data-path $dir/${osd} --op list obj16 | grep \"snapid\":7)"
     ceph-objectstore-tool --data-path $dir/${osd} --rmtype snapmap "$JSON" remove || return 1
     # Check that snapmap is now removed
     ceph-kvstore-tool bluestore-kv $dir/${osd} list 2> /dev/null > $dir/drk.log
-    ! grep "^m.*SNA_.*[.]7[.]obj16[.][.]$" $dir/drk.log || return 1
+    grep SNA_ $dir/drk.log
```
Comment (ronen-fr) [inline qa/standalone/scrub/osd-scrub-snaps.sh:132]: same

**Review by ronen-fr — CHANGES_REQUESTED** [review]

**Review by ronen-fr — COMMENTED** [review]
(How do I change from "request changes" to "Comment"?)

File: qa/standalone/scrub/osd-scrub-snaps.sh (hunk 1 above)
Comment (dzafman) [reply qa/standalone/scrub/osd-scrub-snaps.sh:104]: Yes

File: qa/standalone/scrub/osd-scrub-snaps.sh (hunk 2 above)
Comment (dzafman) [reply qa/standalone/scrub/osd-scrub-snaps.sh:110]: Good idea!

File: qa/standalone/scrub/osd-scrub-snaps.sh (hunk 3 above)
Comment (dzafman) [reply qa/standalone/scrub/osd-scrub-snaps.sh:126]: On failure it allows analysis of what happened.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-01-14_16:22:32-rados-wip-kefu-testing-2021-01-14-2241-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-01-15_15:58:16-rados-wip-kefu-testing-2021-01-15-1726-distro-basic-smithi/

**Comment by neha-ojha** [comment]
@dzafman can you address @ronen-fr's concerns? https://pulpito.ceph.com/kchai-2021-01-14_16:22:32-rados-wip-kefu-testing-2021-01-14-2241-distro-basic-smithi/5786353/ is passing and I'd like to merge it today.

**Comment by dzafman** [comment]
@ronen-fr @neha-ojha Modified the test to accept m or new p DB prefix.  Please approve.

**Review by ronen-fr — APPROVED** [review]

**Comment by neha-ojha** [comment]
testing the new version: https://pulpito.ceph.com/nojha-2021-01-15_21:12:48-rados:standalone-master-distro-basic-smithi/

**Comment by dzafman** [comment]
jenkins test api

**Review by neha-ojha — APPROVED** [review]

**Comment by neha-ojha** [comment]
https://pulpito.ceph.com/nojha-2021-01-15_21:12:48-rados:standalone-master-distro-basic-smithi/ passed - ready to merge

**Pull request merged**